superset-frontend/src/cccs-viz/plugins/plugin-chart-ag-grid/src/ag-grid/ContextMenu/MenuItems/DownloadEmailMenuItem.tsx

```diff
@@ -0,0 +1,97 @@
+import { SupersetClient } from '@superset-ui/core';
+import React from 'react';
+import { saveAs } from 'file-saver';
+import { Menu } from 'src/components/Menu';
+import { useDispatch } from 'react-redux';
+import {
+  addInfoToast,
+  addDangerToast,
+} from 'src/components/MessageToasts/actions';
+import Icon from '@ant-design/icons';
+import EmailSvg from '../../../cccs-grid/images/email.svg';
+
+interface DownloadEmailMenuItemProps {
+  label: string;
+  data: string;
+  onSelection: () => void;
+  key?: string;
+  disabled?: boolean;
+  isContextMenu?: boolean;
+  contextMenuY?: number;
+}
+
+export default function DownloadEmailMenuItem(
+  props: DownloadEmailMenuItemProps,
+) {
+  const dispatch = useDispatch();
+
+  const onClick = () => {
+    const endpoint = `/api/v1/fission/get-eml?file=${props.data}`;
+    dispatch(addInfoToast('Download started'));
+
+    SupersetClient.get({ endpoint, timeout: 180000 }) // 3 minutes
```

Comment:
Can we make a constant for the timeout value?

Reply:
Edited in new commit.